File: frontend_flutter/fruit_market/lib/models/Review.dart
import 'user.dart';
import 'product.dart';

class Review {
  final String reviewId;
  final String userId;
  final String productId;
  final int rating;
  final String? comment;
  final DateTime createdAt;

  final User? user;
  final Product? product;

  Review({
    required this.reviewId,
    required this.userId,
    required this.productId,
    required this.rating,
    this.comment,
    required this.createdAt,
    this.user,
    this.product,
  });

  /// JSON -> Object
  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      reviewId: json['reviewId'],
      userId: json['userId'],
      productId: json['productId'],
      rating: json['rating'],
      comment: json['comment'],
      createdAt: DateTime.parse(json['createdAt']),
      user: json['user'] != null ? User.fromJson(json['user']) : null,
      product:
          json['product'] != null ? Product.fromJson(json['product']) : null,
    );
  }

  /// Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'reviewId': reviewId,
      'userId': userId,
      'productId': productId,
      'rating': rating,
      'comment': comment,
      'createdAt': createdAt.toIso8601String(),
      'user': user?.toJson(),
      'product': product?.toJson(),
    };
  }
}

File: backend_dotnet/fruit_api/fruit_api/Models/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Reviews")]
public class Review
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("reviewId")]  // Sửa từ review_id thành reviewId
    public string ReviewId { get; set; } = null!;

    [Required]
    [Column("userId")]  // Sửa từ user_id thành userId
    public string UserId { get; set; } = null!;

    [Required]
    [Column("productId")]  // Sửa từ product_id thành productId
    public string ProductId { get; set; } = null!;

    [Required]
    [Range(1, 5)]
    [Column("rating")]  // Giữ nguyên
    public int Rating { get; set; }

    [MaxLength(500)]
    [Column("comment")]  // Giữ nguyên
    public string? Comment { get; set; }

    [Column("createdAt")]  // Sửa từ created_at thành createdAt
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    // Navigation properties
    public User? User { get; set; }
    public Product? Product { get; set; }
}